app/models/user_mailbox.rb

```diff
@@ -0,0 +1,44 @@
+class UserMailbox
+  attr_reader :user
+
+  def initialize(user)
+    raise ArgumentError.new("Must provide a user") if user == nil
```

Comment:
This feels overly paranoid to me. Especially since we never catch this error. If I kept it I'd write it as: `unless user`